From db757c3fb8699e79d4867a48ffac3bbe2e1cab71 Mon Sep 17 00:00:00 2001
From: Manu Gupta <mgpta@amazon.com>
Date: Tue, 12 Oct 2021 14:55:52 -0700
Subject: [PATCH] --EKS-PATCH-- ConsistentRead tries 10 times

We've seen clusters where 3 attempts were not enough. Bumping to 10. The
slowdown should be negligible and it will reduce retry attempts in the
upper layers of kubelet.

https://github.com/kubernetes/kubernetes/commit/82cfe9f14f8fb445d682ce2774ea44ce54885e81

This should help (while not fix):
https://t.corp.amazon.com/P51384162/communication

cr: https://code.amazon.com/reviews/CR-58655609
---
 staging/src/k8s.io/mount-utils/mount_helper_unix.go | 2 +-
 vendor/k8s.io/utils/mount/mount_helper_unix.go      | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/staging/src/k8s.io/mount-utils/mount_helper_unix.go b/staging/src/k8s.io/mount-utils/mount_helper_unix.go
index 2c14a27c719..f3658647eea 100644
--- a/staging/src/k8s.io/mount-utils/mount_helper_unix.go
+++ b/staging/src/k8s.io/mount-utils/mount_helper_unix.go
@@ -32,7 +32,7 @@ const (
 	// At least number of fields per line in /proc/<pid>/mountinfo.
 	expectedAtLeastNumFieldsPerMountInfo = 10
 	// How many times to retry for a consistent read of /proc/mounts.
-	maxListTries = 3
+	maxListTries = 10
 )
 
 // IsCorruptedMnt return true if err is about corrupted mount point
diff --git a/vendor/k8s.io/utils/mount/mount_helper_unix.go b/vendor/k8s.io/utils/mount/mount_helper_unix.go
index 11b70ebc298..6110e7b2b35 100644
--- a/vendor/k8s.io/utils/mount/mount_helper_unix.go
+++ b/vendor/k8s.io/utils/mount/mount_helper_unix.go
@@ -32,7 +32,7 @@ const (
 	// At least number of fields per line in /proc/<pid>/mountinfo.
 	expectedAtLeastNumFieldsPerMountInfo = 10
 	// How many times to retry for a consistent read of /proc/mounts.
-	maxListTries = 3
+	maxListTries = 10
 )
 
 // IsCorruptedMnt return true if err is about corrupted mount point
